src/seedu/addressbook/AddressBook.java

```diff
@@ -197,365 +205,392 @@
 
     /*
      * NOTE : =============================================================
-     * Notice how this method solves the whole problem at a very high level.
-     * We can understand the high-level logic of the program by reading this
-     * method alone.
-     * If the reader wants a deeper understanding of the solution, she can go
-     * to the next level of abstraction by reading the methods that are
+     * Notice how this method solves the whole problem at a very high level. We
+     * can understand the high-level logic of the program by reading this method
+     * alone. If the reader wants a deeper understanding of the solution, she
+     * can go to the next level of abstraction by reading the methods that are
      * referenced by the high-level method below.
      * ====================================================================
      */
 
     public static void main(String[] args) {
-        showWelcomeMessage();
-        processProgramArgs(args);
-        loadDataFromStorage();
-        while (true) {
-            String userCommand = getUserInput();
-            echoUserCommand(userCommand);
-            String feedback = executeCommand(userCommand);
-            showResultToUser(feedback);
-        }
+	showWelcomeMessage();
+	processProgramArgs(args);
+	loadDataFromStorage();
+	while (true) {
```

Comment:
as discussed in class, the while-loop can be extracted out into a method so that we apply SLAP to this method.